pandas/_libs/tslibs/strftime.py

```diff
@@ -0,0 +1,278 @@
+        `UnsupportedStrFmtDirective` being raised.
+    target : { "datetime", "date", "time", "period" }, default: "datetime"
+        The kind of data that will be formatted using this template.
+    new_style_fmt : bool, default: False
+        Whether the output string should be new-style
+        e.g. "{year}-{month:02d}-{day:02d} {hour:02d}:{min:02d}:{sec:02d}"
+        or old-style
+        e.g. "%(year)s-%(month)02d-%(day)02d %(hour)02d:%(min)02d:%(sec)02d"
+
+    Returns
+    -------
+    fmt_out : str
+        A string that may be used to format a `datetime` variable. The style of
+        this string is either old-style or new-style depending on
+        `new_style_formatting`.
+        For old-style, it may be used as `fmt_out % fmt_dct`.
+        For new-style, it may be used as `fmt_out.format(**fmt_dct)`
+    locale_dt_strings : LocaleSpecificDtStrings
+        An object containing the locale-specific strings needed for some of the
+        directives. For example locale_dt_strings.am and locale_dt_strings.pm should be
+        used to fill the "ampm" part of the template, induced by directive %p.
+
+    Raises
+    ------
+    UnsupportedStrFmtDirective
+        Raised when the received `strftime_fmt` format contains a directive for
+        which the output can not currently be created using string formatting.
+
+    See Also
+    --------
+    `strftime format codes reference
+    <https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format\
+-codes>`_
+
+    `Stackoverflow post <https://stackoverflow.com/a/43495629/7262247>`_
+    explaining how old-style formatting is faster than new-style formatting,
+    itself faster than datetime.strftime`.
+
+    See `Period.strftime` doc for all the supported period directives (same
+    directives as the :func:`time.strftime` function of the standard Python
+    distribution, as well as specific additional directives ``%f``, ``%F``,
+    ``%q``, ``%l``, ``%u``, ``%n``).
+    """
+    unsupported: tuple[tuple[str, ...], ...]
+    if target in ("datetime", "date", "time"):
+        directive_maps = (_COMMON_MAP, _DATETIME_MAP)
+        unsupported = (_COMMON_UNSUPPORTED, _DATETIME_UNSUPPORTED)
+    elif target == "period":
+        directive_maps = (_COMMON_MAP, _PERIOD_MAP)
+        unsupported = (_COMMON_UNSUPPORTED, _PERIOD_UNSUPPORTED)
+    else:
+        raise ValueError(f"Invalid target: {target!r}")
+
+    # Raise if unsupported directive found in `strftime_fmt`
+    for _u in unsupported:
+        for key in _u:
+            if key in strftime_fmt:
+                raise UnsupportedStrFmtDirective(f"Unsupported directive: '{key}'")
+
+    # Mapping between strftime and string formatting, according to both styles
+    if new_style_fmt:
+        esc = "/_+\\"
+
+        # Escape the %% before searching for directives, same as strftime
+        strftime_fmt = strftime_fmt.replace("%%", esc)
+
+        esc_l = "+^_\\"
+        esc_r = "/_^+"
```

Comment:
this looks a bit mysterious to me, mind explaining why you've constructed `esc`, `esc_l` and `esc_r` like this?

Reply:
haha good catch. Indeed this is ugly. Purely random strings here. Long enough so that users will not hit them by chance. 

A more robust way would probably be to check that the above do not exist in the input sequence, and if they do, to add as many trailing '+' (or any other char) as needed so that it is not present in the input string. We could implement it in a `find_escape_patterns` internal function. 
If you agree with this strategy I can change it like this.

Reply:
fixed